Add new CFG Array version (5.5)

klarich opened the pull request:
Changes include: 

- Created a new function called clean_glycan_string to assist with removing small errors from CFG glycan strings
- Added an example script for generating a new glycan array file
- Updated README to describe the process for adding a new glycan array file 
- Added new glycan array version csv's to `Data/CFG_Array_Versions` and `ccarl/glycan_parsers/CFG_Array_Versions` - note: I'm not sure if it is necessary to add the csv to both locations or if it only needs to be added to one of those locations, however, since all the csv files were in both locations I added it to both. 

I have not built the Dockerfile or tested the code yet as I was running into some errors building the image on my machine - but I can dig into this a bit more.

kklarich committed:
Add CFG array version 5.5

kklarich committed:
Add CFG array version 5.5

klarich committed:
Add CONTRIBUTING.md

klarich committed:
Add CONTRIBUTING.md

klarich committed:
remove CONTRIBUTING.md

klarich committed:
remove CONTRIBUTING.md

klarich committed:
Add helper function clean_glycan_string

klarich committed:
Add helper function clean_glycan_string

klarich committed:
Update README

klarich committed:
Update README

klarich committed:
Add example script for generating new glycan array version

klarich committed:
Add example script for generating new glycan array version

klarich committed:
Add new glycan array version

klarich committed:
Add new glycan array version

klarich committed:
Update README

klarich committed:
Update README

andrewguy commented on Scripts/generate_new_glycan_array_version.py:
```diff
@@ -0,0 +1,36 @@
+        print(structure)
+
+    if not failed:  # Only write new glycan array file if all structures were successfully parsed.
+        with open(output_path, 'w') as f:
+            for chart_number, structure in zip(df['Chart Number'], succeeded):
+                f.write(f"{chart_number},{structure}\n")
```
```suggestion
        df['Structure'] = succeeded
        df.to_csv(output_path, columns=["Chart Number", "Structure"], index=False, header=False)
```

Using pandas to write to csv will handle any issues with escaping `,` characters.

klarich replied:
Ah good catch

andrewguy commented on ccarl/glycan_parsers/cfg_array_versions.py:
```diff
@@ -55,3 +53,17 @@ def get_likely_cfg_array_version(glycan_list, distance_threshold=2.0):
+
+def clean_glycan_string(glycan: str) -> str:
+    """
+    Method to clean glycan strings before parsing. n
```
```suggestion
    Method to clean glycan strings before parsing.
    
```

andrewguy commented on Data/CFG_Array_Versions/Array_5.5.csv:
```diff
@@ -0,0 +1,562 @@
+45,(6S)Galb1-4(6S)Glcb-Sp8
+46,Neu5Aca2-3(6S)Galb1-4GlcNAcb-Sp8
+47,(6S)GlcNAcb-Sp8
+48,Neu5,9Ac2a-Sp8
```
```suggestion
48,"Neu5,9Ac2a-Sp8"
```

andrewguy commented on Data/CFG_Array_Versions/Array_5.5.csv:
```diff
@@ -0,0 +1,562 @@
+46,Neu5Aca2-3(6S)Galb1-4GlcNAcb-Sp8
+47,(6S)GlcNAcb-Sp8
+48,Neu5,9Ac2a-Sp8
+49,Neu5,9Ac2a2-6Galb1-4GlcNAcb-Sp8
```
```suggestion
49,"Neu5,9Ac2a2-6Galb1-4GlcNAcb-Sp8"
```

andrewguy commented on Data/CFG_Array_Versions/Array_5.5.csv:
```diff
@@ -0,0 +1,562 @@
+315,Neu5Aca2-3Galb1-4GlcNAcb1-2Mana1-6(Neu5Aca2-3Galb1-4GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp12
+316,Neu5Aca2-3Galb1-4GlcNAcb1-2Mana1-6(Neu5Aca2-6Galb1-4GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp12
+317,Galb1-4(Fuca1-3)GlcNAcb1-2Mana1-6(Galb1-4(Fuca1-3)GlcNAcb1-2Mana1-3)Manb1-4GlcNAcb1-4GlcNAcb-Sp20
+318,Neu5,9Ac2a2-3Galb1-3GlcNAcb-Sp0
```
```suggestion
318,"Neu5,9Ac2a2-3Galb1-3GlcNAcb-Sp0"
```

andrewguy committed:
Update dockerfile to use code in current repo

klarich committed:
Use pandas to write to csv

Co-authored-by: Andrew Guy <andrewjguy42@gmail.com>

klarich committed:
Use pandas to write to csv

Co-authored-by: Andrew Guy <andrewjguy42@gmail.com>

klarich committed:
Fix typo

Co-authored-by: Andrew Guy <andrewjguy42@gmail.com>

klarich committed:
Fix typo

Co-authored-by: Andrew Guy <andrewjguy42@gmail.com>

klarich committed:
Fix glycan string with comma

Co-authored-by: Andrew Guy <andrewjguy42@gmail.com>

klarich committed:
Fix glycan string with comma

Co-authored-by: Andrew Guy <andrewjguy42@gmail.com>

klarich committed:
Regenerate csv files using pandas

klarich committed:
Regenerate csv files using pandas

klarich committed:
Small variable name fixes

klarich committed:
Small variable name fixes

klarich committed:
New line at end of file

This keeps getting automatically added everytime I try to git commit, preventing me from rebasing

klarich committed:
New line at end of file

This keeps getting automatically added everytime I try to git commit, preventing me from rebasing

klarich committed:
Merge remote-tracking branch 'origin/master'
